### PR TITLE
Update definition of optimizer in introduction_guide.rst

### PR DESCRIPTION
**Pull request by Code-Cornelius**
I looked at the guidelines and this pull request does not fix a bug, but simply make sure that the tutorial run with the code base from the version 1.5.3. Hence, I do not think it is useful to create another branch for this.

Please let me know if I have missed any guidelines.

sorry that I closed / reopened the issue, it is my first pull request.

## What does this PR do?

The guide was a bit misleading, and following it naivly would make that the code is not runnable, because the training requires an optimisers. My solution is to put the optimiser chunk higher in the explainations.

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/starter/introduction_guide.rst [hunk 1]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
     torch.Size([1, 10])
 
 
-Now we add the training_step which has all our training loop logic
+Now we add the training_step to the previously defined class which has all our training loop logic
```
Comment (awaelchli) [inline docs/source/starter/introduction_guide.rst:119]: I think it is clear from the context where to add it to

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Just noticed, there is a PR already trying to achieve mostly the same thing: #10519

File: docs/source/starter/introduction_guide.rst [hunk 2]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
             x, y = batch
             logits = self(x)
             loss = F.nll_loss(logits, y)
             return loss
 
+
+Optimizer
+---------
+
+Next we choose what optimizer to use for training our system.
```
Comment (tchaton) [inline docs/source/starter/introduction_guide.rst:136]:
```suggestion
Next, we choose which optimizer to use for training our system.
```

File: docs/source/starter/introduction_guide.rst [hunk 3]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
+In PyTorch we do it as follows:
```
Comment (tchaton) [inline docs/source/starter/introduction_guide.rst:137]:
```suggestion
In PyTorch, we do it as follows:
```

File: docs/source/starter/introduction_guide.rst [hunk 4]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
+
+.. code-block:: python
+
+    from torch.optim import Adam
+
+    optimizer = Adam(LitMNIST().parameters(), lr=1e-3)
+
+
+In Lightning we do the same but organize it under the :func:`~pytorch_lightning.core.LightningModule.configure_optimizers` method.
```
Comment (tchaton) [inline docs/source/starter/introduction_guide.rst:146]:
```suggestion
In Lightning, the same code is re-organized within the :func:`~pytorch_lightning.core.LightningModule.configure_optimizers` method.
```

File: docs/source/starter/introduction_guide.rst [hunk 5]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
+
+.. testcode::
+
+    class LitMNIST(LightningModule):
+        def configure_optimizers(self):
+            return Adam(self.parameters(), lr=1e-3)
+
+.. note:: The LightningModule itself has the parameters, so pass in self.parameters()
+
+However, if you have multiple optimizers use the matching parameters
```
Comment (tchaton) [inline docs/source/starter/introduction_guide.rst:156]:
```suggestion
If you have multiple optimizers, simply return them all.
```

File: docs/source/starter/introduction_guide.rst [hunk 6]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
+
+.. testcode::
+
+    class LitMNIST(LightningModule):
+        def configure_optimizers(self):
+            return Adam(self.generator(), lr=1e-3), Adam(self.discriminator(), lr=1e-3)
+
+
```
Comment (tchaton) [inline docs/source/starter/introduction_guide.rst:164]: example with a scheduler?

File: docs/source/starter/introduction_guide.rst (hunk 4 above)
Comment (rohitgr7) [inline docs/source/starter/introduction_guide.rst:146]:
```suggestion
In Lightning we do the same but organize it under the :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_optimizers` method.
```

File: docs/source/starter/introduction_guide.rst [hunk 7]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
+In Lightning we do the same but organize it under the :func:`~pytorch_lightning.core.LightningModule.configure_optimizers` method.
+
+.. testcode::
+
+    class LitMNIST(LightningModule):
+        def configure_optimizers(self):
+            return Adam(self.parameters(), lr=1e-3)
+
+.. note:: The LightningModule itself has the parameters, so pass in self.parameters()
```
Comment (rohitgr7) [inline docs/source/starter/introduction_guide.rst:154]:
```suggestion
.. note:: The ``LightningModule`` itself has the parameters, so pass in ``self.parameters()``.
```

File: docs/source/starter/introduction_guide.rst [hunk 8]
```diff
@@ -116,17 +116,53 @@ equivalent to a pure PyTorch Module except it has added functionality. However,
...
+
+However, if you have multiple optimizers use the matching parameters
+
+.. testcode::
+
+    class LitMNIST(LightningModule):
+        def configure_optimizers(self):
+            return Adam(self.generator(), lr=1e-3), Adam(self.discriminator(), lr=1e-3)
```
Comment (rohitgr7) [inline docs/source/starter/introduction_guide.rst:162]:
```suggestion
            return Adam(self.generator.parameters(), lr=1e-3), Adam(self.discriminator.parameters(), lr=1e-3)
```

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: docs/source/starter/introduction_guide.rst [hunk 9]
```diff
@@ -146,23 +147,40 @@ In PyTorch, the optimizer is created as follows:
...
+
+.. testcode::
+
+    from torch.optim.lr_scheduler import CosineAnnealing
+
+
+    class LitMNIST(LightningModule):
+        def configure_optimizers(self):
+            opt = Adam(self.parameters(), lr=1e-3)
+            scheduler = CosineAnnealing(opt, T_max=10)
```
Comment (rohitgr7) [inline docs/source/starter/introduction_guide.rst:178]:
```suggestion
    from torch.optim.lr_scheduler import CosineAnnealingLR


    class LitMNIST(LightningModule):
        def configure_optimizers(self):
            opt = Adam(self.parameters(), lr=1e-3)
            scheduler = CosineAnnealingLR(opt, T_max=10)
```
